### DISCUSSION TITLE
Search preview ≠ search deploy

### DISCUSSION THREAD

**Discussion by Martien** [observablehq/framework#1888] [Q&A] [answered]
During preview, search does find a page, while the built and deployed version does not.

For example, in preview I search for ‘toeval’ and it correctly lists http://127.0.0.1:3000/toeval-bestaat. Next, I build and deploy the site, but once live, it finds different pages, and not the top one shown in the preview instance.

Is there a way to check/test minisearch?

Demo: search for ‘**toeval**’ on http://aarde.aardrock.nl/. Top hit should be http://aarde.aardrock.nl/toeval-bestaat (as in preview), but this page is not even listed.

Confused.

**Comment by Fil** [comment]
Is this different from https://github.com/observablehq/framework/discussions/1882 ?

**Reply by Martien** [reply]
Yes it is. I have added `index: true` to all pages so it finds pages beyond level 2.

In preview, I get what I search for. In deployed, I get different and unexpected results, as mentioned above. Try the demo above and be surprised that `toeval bestaat` is not top of the list; not even in the list, when searching for “toeval”.

Page source: https://github.com/Martien/aarde/blob/main/src//toeval-bestaat.md?plain=1

**Reply by Martien** [reply]
For reference: **_preview_** shows this when entering search string ‘toeval’:
<img width="1178" alt="Screenshot 2024-12-22 at 11 38 56" src="https://github.com/user-attachments/assets/4404dc6b-0ff6-401d-ad55-753dbb5c7f5e" />

**Reply by Martien** [reply]
And this on the **_deployed_** version:
<img width="1178" alt="Screenshot 2024-12-22 at 11 41 41" src="https://github.com/user-attachments/assets/75ba81db-fe76-46f9-836f-c404b59a3940" />

**Reply by Fil** [reply]
Worked for me: I cloned the repo, changed **output** to be local on my machine, and did `npm run build` then `npx http-serve ./nl.aardrock.aarde/2024-12-22T12:24:16.799/`.

<img width="416" alt="Capture d’écran 2024-12-22 à 12 27 50" src="https://github.com/user-attachments/assets/b87c37eb-1ace-4b33-8cbc-8df385a5e448" />

**Reply by Martien** [reply]
Okay, weird. Will double check and clean installs, caches, etc.

**Reply by Martien** [reply]
And thanks for trying!

**Reply by Martien** [reply, marked as answer]
🙏 So sorry. I got bitten by the Symlink Caching Monster: macOS and some network protocols (like SMB or NFS) cache symlink paths to optimize performance, which can cause delays in reflecting updates to symlinks.

My webserver was still using a link to the previous instance.

[curled up in shame in a corner of my screen]
